### PR TITLE
Add travis.yml and image deploy to bintray

### PR DESCRIPTION
**Pull request by jlongever**
- deploy microkernel and ipxe build artifacts to bintray
- Includes some ansible tweaks

@RackHD/corecommitters @RackHD/rackhd_dev

### CODE REVIEW

**Comment by heckj** [comment]
+1

File: .bintray.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+    /* Bintray package information.
+       In case the package already exists on Bintray, only the name, repo and subject
+       fields are mandatory. */
```
Comment (benbp) [inline .bintray.json]: Does bintray have their own json parser or something to handle comments?

File: common/tasks/mount_overlay.yml [hunk 2]
```diff
@@ -12,10 +12,20 @@
     - "{{ overlayfs_lower_dir }}"
     - "{{ overlayfs_upper_dir }}"
     - "{{ overlayfs_build_root }}"
+    - "{{ overlayfs_work_dir }}"
 
-- name: Mount overlay
+- name: Mount overlayfs
+  when: kernel_major == "3" and kernel_minor == "16"
```
Comment (benbp) [inline common/tasks/mount_overlay.yml]: What about 3.13 (since that's what the travis file is having us use)?

File: group_vars/on_imagebuild [hunk 3]
```diff
@@ -21,3 +21,7 @@ repository_lists:
 # Referenced by basefs and overlay roles
 squashfs_image_path: "{{ build_artifact_path }}/{{ basefs_image_name }}"
 basefs_image_name: "base.{{ debootstrap_release }}.{{ ubuntu_kernel_version }}.squashfs.img"
+
+# Local running kernel major.minor version
+kernel_major: "{{ ansible_kernel.split('.')[0] }}"
```
Comment (benbp) [inline group_vars/on_imagebuild:26]: Since we can call split, can we also call toInt or is that too much to ask for?

Comment (benbp) [inline group_vars/on_imagebuild:26]: Boom:

```
when: kernel_major | int <= 16
```

via: https://groups.google.com/forum/#!topic/ansible-project/kzzsYro1Z1c

File: common/tasks/mount_overlay.yml (hunk 2 above)
Comment (jlongever) [inline common/tasks/mount_overlay.yml]: The target kernel version for the discovery image is 3.13,  these checks are for the travis hosts running kernel version.

File: group_vars/on_imagebuild (hunk 3 above)
Comment (jlongever) [inline group_vars/on_imagebuild:26]: Right,  that did not work when I tried it.  Wasn't sure if it was an ansible version thing.  I can try it again though.

File: .bintray.json (hunk 1 above)
Comment (jlongever) [inline .bintray.json]: I suppose so since it parses without any errors... the comments here were from an example.

Comment (benbp) [inline .bintray.json]: Hm, they probably just strip comments before parsing it, actually.

File: common/tasks/mount_overlay.yml (hunk 2 above)
Comment (benbp) [inline common/tasks/mount_overlay.yml]: oh, duh (smacks forehead)

File: group_vars/on_imagebuild (hunk 3 above)
Comment (benbp) [inline group_vars/on_imagebuild:26]: oh, yeah probably since we're stuck on 1.5. Bleh.

Comment (jlongever) [inline group_vars/on_imagebuild:26]: guess i didn't try that hard,  updated.

Comment (benbp) [inline group_vars/on_imagebuild:26]: Oh, haha

**Comment by benbp** [comment]
:+1:

**Comment by benbp** [comment]
yay!

**Comment by jlongever** [comment]
updated,  enabled file override in bintray.

**Comment by benbp** [comment]
Still :+1:

**Pull request merged**